scripts/prepare_heat_demand.py

```diff
@@ -0,0 +1,305 @@
+    given scenario
+
+    Parameters
+    ----------
+    path : str
+        input path
+    region : str
+        region e.g. BB as Brandenburg
+    scenario : str
+        scenario e.g. "base"
+
+    Returns
+    -------
+    demand_ghd : float
+        total demand ghd
+    demand_hh : float
+        total demand hh
+    demand_ghd_unit : str
+        unit of total demands
+
+    """
+    # Read state heat demands of ghd and hh sectors
+    sc = dp.load_b3_scalars(path)  # TODO: Add ', sep=";"' if from schema
+    consumers = ["ghd", "efh", "mfh"]
+    types = ["central", "decentral"]
+    demands = pd.DataFrame()
+    units = []
+
+    for row in sc.iterrows():
+        # Get heat demands in region
+        if (
+            "heat" in row[1]["carrier"]
+            and "demand" in row[1]["tech"]
+            and row[1]["scenario"] == scenario
+            and row[1]["region"] == region
+        ):
+            for consumer in consumers:
+                for type in types:
+                    if (
+                        consumer in row[1]["name"]
+                        and type in row[1]["name"]
+                        and "de" + type not in row[1]["name"]
+                    ):
+                        check_central_decentral(
+                            demands, row[1]["var_value"], consumer, type
+                        )
+                        units.append(row[1]["var_unit"])
+
+    if len(list(set(units))) != 1:
+        raise ValueError(
+            f"Unit mismatch in scalar data of heat demands. "
+            f"Please make sure units match in {path}."
+        )
+
+    return demands, list(set(units))
+
+
+def calculate_heat_load(region, scenario):
+    """
+    This function calculates a heat load profile of Industry, trade,
+    service (ghd: Gewerbe, Handel, Dienstleistung) and Household (hh: Haushalt)
+    sectors
+
+    Parameters
+    ----------
+    region : str
+        region e.g. BB as Brandenburg
+    scenario : str
+        scenario e.g. "base"
+
+    Returns
+    -------
+    heat_load_total : pd.DataFrame
+         DataFrame that contains the calculated total heat load
+
+    """
+    # Set examined years
+    years = np.arange(2010, 2020)
+
+    # Read time series template
+    ts_template = dp.load_b3_timeseries(in_path5, sep=";")
+
+    # Add empty data frame for results / output
+    heat_load = pd.DataFrame(columns=ts_template.columns)
+
+    for index, year in enumerate(years):
+        # Get holidays
+        holidays = get_holidays(in_path3, year, region)
+
+        # Read weather data
+        path_weather_data = os.path.join(
+            in_path1, "weatherdata_" + region + "_" + str(year) + ".csv"
+        )
+        temperature = pd.read_csv(path_weather_data, usecols=["temp_air"], header=0)
+
+        # Get heat demand in region and scenario
+        yearly_demands, sc_demand_unit = get_heat_demand(in_path4, region, scenario)
+
+        types = ["central", "decentral"]
```

Comment:
`heat_central` and `heat_decentral` are carriers in our definition, not types.

Reply:
Done with commit https://github.com/rl-institut/oemof-B3/pull/76/commits/ae095e268fdc66e2277c389abf1a33e7abe29270.